### PR TITLE
Long/update user status and last message

### PR DESCRIPTION
**Pull request by khuolo01**
Ticket: https://tan-tho-le.atlassian.net/jira/software/projects/VAJ/boards/1?selectedIssue=VAJ-56

This PR implements:

- API to fetch last_message.
- Updating conversation's last_message and otherParticipantStatus in real-time using WebSocket.
- Adding a child component for ConversationCard called LastMessage.

New Design:
Component

### CODE REVIEW

**Comment by khuolo01** [comment]
This PR should be on hold until we finish merging @vuongtranluc's [PR](https://github.com/leth02/VL_Chat_App/pull/30).

**Comment by khuolo01** [comment]
@leth02 @vuongtranluc We can start reviewing this PR.

File: message_app_server/message_app/send_message.py
Comment (leth02) [inline message_app_server/message_app/send_message.py:12]: Is there any difference between setting the CORS here and this?
`CORS(app, origins=["http://localhost:3000"])`

File: message_app_server/message_app/send_message.py
Comment (leth02) [inline message_app_server/message_app/send_message.py]: I thought that we can combine all of the endpoints that handle "fetching messages" into one endpoint, with an additional param `options` that specify what needs to be fetched. But we can leave it for later.

File: message_app_server/message_app/model.py
Comment (leth02) [inline message_app_server/message_app/model.py:71]: So here we reason that a user will be uniquely identified with tuple `(username, id, email)` ?

File: message_app_client/src/components/conversation.js
Comment (leth02) [inline message_app_client/src/components/conversation.js:47]: It's good that you understand how to use this 👍

File: message_app_server/message_app/send_message.py
Comment (khuolo01) [reply message_app_server/message_app/send_message.py:12]: Yes. `CORS(app, origins=["http://localhost:3000"])` is for the client fetching data through API, while this one is for WebSocket. They are not related to each other so we have to set up cors for both.

I passed `'http://localhost:5000'` to the whitelist so the old version can still work. It will be deleted after we finish migrating.

File: message_app_server/message_app/model.py
Comment (khuolo01) [reply message_app_server/message_app/model.py:71]: Yes, but we only need to pass one of them to get the user from the database.

File: message_app_server/message_app/send_message.py
Comment (khuolo01) [reply message_app_server/message_app/send_message.py]: Yes. Also, this one is actually not just for getting the last message but for getting any message with a specific id.

File: message_app_server/message_app/send_message.py
Comment (leth02) [reply message_app_server/message_app/send_message.py:12]: What do you mean by old version? Old version of WebSocket?

Also, please add a note here about what you just said
`// whitelist 'http://localhost:5000' to keep old WebSocket works. It will be deleted after we finish migrating`

File: message_app_server/message_app/model.py
Comment (leth02) [reply message_app_server/message_app/model.py:71]: I do believe that the only thing that we should use is the ID, because we may allow users to change their email and probably username.

File: message_app_server/message_app/send_message.py
Comment (khuolo01) [reply message_app_server/message_app/send_message.py:12]: The one using plain javascript (the file `socketio.js`). Its domain is `http://localhost:5000`, so I have to put that domain into the whitelist to make that version works.

Comment (leth02) [reply message_app_server/message_app/send_message.py:12]: Gotcha. thanks.

File: message_app_server/message_app/model.py
Comment (khuolo01) [reply message_app_server/message_app/model.py:71]: We let users log in with their username or email address, so we need to query with username and email address too. Also, I won't think using any of them will cause trouble because all three columns are unique.

**Review by leth02 — APPROVED** [review]

**Pull request merged**
